flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09Fetcher.java

```diff
@@ -283,10 +296,16 @@ public void commitSpecificOffsetsToKafka(Map<KafkaTopicPartition, Long> offsets)
 			}
 		}
 
-		if (this.consumer != null) {
-			synchronized (consumerLock) {
-				this.consumer.commitSync(offsetsToCommit);
-			}
+		if (commitInProgress) {
+			LOG.warn("Committing offsets to Kafka takes longer than the checkpoint interval. " +
+					"Some checkpoints may be subsumed before committed. " +
+					"This does not compromise Flink's checkpoint integrity.");
+		}
```

Comment:
I would not want to put it into the main loop, because then the warning would come repeatedly, every time the poll happens.